plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerInstanceProvider.java

```diff
@@ -239,9 +241,9 @@ public Instance createInstance(InstanceKey instanceKey,
                                    Machine machine,
                                    LineConsumer creationLogsOutput) throws NotFoundException, MachineException {
         final DockerInstanceKey dockerInstanceKey = new DockerInstanceKey(instanceKey);
-
-        pullImage(dockerInstanceKey, creationLogsOutput);
-
+        if (snapshotUseRegistry) {
```

Comment:
Value of this property can be changed between calls of snapshot saving and recovering.